model/impl.go

```diff
@@ -80,6 +85,21 @@ func (rc *realModel) addNode(hostname string) *NodeInfo {
 	return node_ptr
 }
 
+// Deletes the given node with all of the pods and containers on it.
+func (rc *realModel) deleteNode(hostname string) {
+	rc.lock.Lock()
+	defer rc.lock.Unlock()
+
+	if node, ok := rc.Nodes[hostname]; ok {
+		for _, podInfo := range node.Pods {
+			if namespaceInfo, ok := rc.Namespaces[podInfo.Namespace]; ok {
```

Comment:
Both deleteNode and pod operate on the same lock. Here the lock is already taken and deletePod would deadlock (at least i think so - mutex doc doesn't say anything about possibility to acquire a lock multiple times - "Lock locks m. If the lock is already in use, the calling goroutine blocks until the mutex is available.")